src/reducers/columnStateHelper.js

```diff
@@ -29,15 +30,9 @@ const DRAG_SCROLL_BUFFER = 100;
  * @return {!Object}
  */
 function initialize(state, props, oldProps) {
-  const { scrollLeft, scrollToColumn } = props;
   let { columnResizingData, isColumnResizing, scrollX } = state;
 
-  if (scrollLeft !== undefined &&
-      (!oldProps || scrollLeft !== oldProps.scrollLeft)) {
-    scrollX = scrollLeft;
-  }
-
-  scrollX = scrollTo(state, props, oldProps.scrollToColumn, scrollX);
+  const columnAnchor = getColumnAnchor(state, props, oldProps);
 
   const { maxScrollX } = columnWidths(state);
   scrollX = clamp(scrollX, 0, maxScrollX);
```

Comment:
should this clamping be moved somewhere further out so it occurs after the scrollAnchor is converted into the scrollX?

Reply:
> should this clamping be moved somewhere further out so it occurs after the scrollAnchor is converted into the scrollX?

pretty sure we can

the same constraint is now applied at the end of `computeRenderedColumns`